### ISSUE TITLE
fix: lr_scheduler.step() called before optimizer.step() warning with AMP training

### ISSUE THREAD

**Issue by yukiharada1228** [yukiharada1228/KnowledgeTransferGraph#5]
## 問題

AMP（混合精度）学習中に以下の PyTorch 警告が発生しています。

```
UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`.
In PyTorch 1.1.0 and later, you should call them in the opposite order:
`optimizer.step()` before `lr_scheduler.step()`.
```

## 原因

`dml/graph.py` の `Node.optimize()` と `Graph.step_schedulers()` の連携に問題があります。

```python
# graph.py:154-158
def optimize(self, loss: torch.Tensor):
    self.optimizer.zero_grad()
    self.scaler.scale(loss).backward()
    self.scaler.step(self.optimizer)  # ← AMP overflow 時はスキップされる
    self.scaler.update()
```

```python
# trainer.py:66-67
self.graph.optimize_all(losses)
self.graph.step_schedulers("step")  # ← 常に呼ばれる（optimizer.step() がスキップされた場合も）
```

AMP 学習では `GradScaler` が勾配の overflow（NaN/inf）を検出した場合、`optimizer.step()` を **スキップ** します。
しかし `scheduler.step()` は `optimize_all()` の直後に無条件で呼ばれるため、
optimizer が実際には step していないのに scheduler だけが step するケースが発生します。

これにより PyTorch の内部カウンタ（`optimizer._step_count` vs `scheduler._step_count`）がずれ、警告が発生します。

## 再現条件

- `Node.scaler` に `GradScaler` を設定した AMP 学習
- `scheduler_interval="step"` または `scheduler_interval="epoch"` 問わず発生しうる
- 訓練初期（fp16 で overflow が起きやすい段階）に特に発生

## 修正方針

`Node.optimize()` でオプティマイザが実際に step したかどうかを返し、`step_schedulers()` がそれを受け取って条件付きで `scheduler.step()` を呼ぶよう修正する。

```python
# 修正案
def optimize(self, loss: torch.Tensor) -> bool:
    """Returns True if optimizer actually stepped (no gradient overflow)."""
    self.optimizer.zero_grad()
    self.scaler.scale(loss).backward()
    old_scale = self.scaler.get_scale()
    self.scaler.step(self.optimizer)
    self.scaler.update()
    return self.scaler.get_scale() >= old_scale  # overflow なければ True
```

`Graph.optimize_all()` → `Graph.step_schedulers()` にこの情報を伝播させる。

## 関連ファイル

- `dml/graph.py`: `Node.optimize()`, `Graph.optimize_all()`, `Graph.step_schedulers()`
- `dml/trainer.py`: `Trainer.fit()`

**Comment by yukiharada1228** [comment]
実際の学習への影響がほぼ皆無であり、修正によるコードの複雑化のコストが見合わないため、対応しないこととします。
